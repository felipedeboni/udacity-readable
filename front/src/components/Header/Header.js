import React from 'react';
import { Link, NavLink } from 'react-router-dom';
import react from 'react.svg';
import './Header.css';

const Header = ({ categories = [] }) => {
  return (
    <header className="app__header">
      <Link to="/" className="app__logo">
        <img src={react} alt="none" /> Readable
      </Link>
      <nav className="app__nav">
        <ul className="list-unstyled">
          <li className="nav__item nav__item--new-post">
            <NavLink to="/posts/new" className="nav__link">
              + Post
            </NavLink>
          </li>
          {categories.map((category, i) => (
            <li key={i} className="nav__item nav__item--sections">
              <NavLink to={`/${category.path}`} className="nav__link">
                {category.name}
              </NavLink>
            </li>
          ))}
        </ul>
      </nav>
    </header>
  );
};

export default Header;
