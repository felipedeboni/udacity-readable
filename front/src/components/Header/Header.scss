.app__header {
  border-bottom: 3px solid #F3F3F3;
  padding: 10px 0;
  display: flex;
  flex-direction: row;
  align-items: center;
  margin-bottom: 20px;
}

.app__logo {
  display: flex;
  align-items: center;
  font-size: 18px;

  &:hover {
    text-decoration: none;
  }

  img {
    height: 40px;
  }
}

.app__nav {
  display: flex;
  justify-content: flex-end;
  width: 100%;

  ul {
    align-items: center;
    display: flex;
    margin: 0;
    padding: 0;
  }
}

.nav__link {
  display: flex;
  font-size: 18px;
  font-weight: 700;

  &:hover,
  &:focus {
    text-decoration: none;
  }
}

.nav__item--new-post {
  margin-right: 15px;
}

.nav__item--sections {
  border-style: solid;
  border-color: #F3F3F3;
  border-width: 2px 0 2px 2px;
  color: #8F8F8F;
  text-transform: capitalize;

  .nav__link {
    color: #8F8F8F;
    padding: 4px 10px;

    &.active {
      background: #F3F3F3;
      color: #54595F;
    }
  }

  &:last-child {
    border-right-width: 2px;
  }
}
