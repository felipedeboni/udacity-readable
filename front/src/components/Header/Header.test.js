import React from 'react';
import Header from './Header';
import { shallow } from 'enzyme';

describe('Header', () => {
  it('renders without crashing', () => {
    const test = <Header />;

    expect(shallow(test)).toMatchSnapshot();
  });

  it('renders without crashing with categories', () => {
    const { categories } = global.mock;

    const test = <Header categories={categories} />;

    expect(shallow(test)).toMatchSnapshot();
  });
});
