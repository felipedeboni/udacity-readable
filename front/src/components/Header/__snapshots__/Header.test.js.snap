// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders without crashing 1`] = `
<header
  className="app__header"
>
  <Link
    className="app__logo"
    replace={false}
    to="/"
  >
    <img
      alt="none"
      src="react.svg"
    />
     Readable
  </Link>
  <nav
    className="app__nav"
  >
    <ul
      className="list-unstyled"
    >
      <li
        className="nav__item nav__item--new-post"
      >
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          className="nav__link"
          to="/posts/new"
        >
          + Post
        </NavLink>
      </li>
    </ul>
  </nav>
</header>
`;

exports[`Header renders without crashing with categories 1`] = `
<header
  className="app__header"
>
  <Link
    className="app__logo"
    replace={false}
    to="/"
  >
    <img
      alt="none"
      src="react.svg"
    />
     Readable
  </Link>
  <nav
    className="app__nav"
  >
    <ul
      className="list-unstyled"
    >
      <li
        className="nav__item nav__item--new-post"
      >
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          className="nav__link"
          to="/posts/new"
        >
          + Post
        </NavLink>
      </li>
      <li
        className="nav__item nav__item--sections"
        key="0"
      >
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          className="nav__link"
          to="/react"
        >
          react
        </NavLink>
      </li>
      <li
        className="nav__item nav__item--sections"
        key="1"
      >
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          className="nav__link"
          to="/redux"
        >
          redux
        </NavLink>
      </li>
      <li
        className="nav__item nav__item--sections"
        key="2"
      >
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          className="nav__link"
          to="/udacity"
        >
          udacity
        </NavLink>
      </li>
    </ul>
  </nav>
</header>
`;
